pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -159,6 +161,15 @@ func (t ResolvedPipelineRunTask) IsSkipped(state PipelineRunState, d *dag.Graph)
 		}
 	}
 
+	// Check if the when expressions have been
+	if len(t.PipelineTask.WhenExpressions) > 0 {
+		if haveVariables := t.PipelineTask.WhenExpressions.HaveVariables(); !haveVariables {
+			if areTrue := t.PipelineTask.WhenExpressions.AreTrue(); !areTrue {
+				return true
```

Comment:
this will be slightly clearer as:

```go
if ! t.PipelineTask.WhenExpressions.AreTrue() {
```

Reply:
I'm also really tempted to suggest a different name for AreTrue, something like: `ShouldExecute` or `AreSatisfied`, (maybe that second one isnt any better, i couldnt think of much besides ShouldExecute XD)

they're already called "whenexpressions" so the "when" part kind indicates they are for something specific, i.e. determining WHEN to execute something, so i think it makes sense to name their functions a bit more after what they're for vs. the details of what they do

i.e. maybe AreTrue does check if all the expressions are true, but is it for? its for indicating if something should execute

so then this code could be:

```
if ! t.PipelineTask.WhenExpressions.ShouldExecute() {
```

oh oh oh or here's another idea, invert the logic and make the function `BlockExecution` or `ShouldBlock` or `Blocks`, then you' get something like:

```
if t.PipelineTask.WhenExpressions.Blocks() {
```

Reply:
these are great ideas...we could also use `Allows` for allows execution of the guarded `Task`
```
if t.PipelineTask.WhenExpressions.Allows() {
```